runner/full.go

```diff
@@ -229,7 +238,7 @@ func CollectAllServers(ctx context.Context, servers []*state.Server, globalColle
 					}
 				}
 			} else {
-				server.Grant = grant
+				server.Grant.Store(&grant)
```

Comment:
As discussed, this might be a problem in the websocket case because `grant` could be the previous grant, and `server.Grant` might already have been replaced by a newer grant. I think it seems safe to just not do this store here (and in the error case 12 lines up) if we're on a websocket connection.